pkg/cmd/pipeline/create.go

```diff
@@ -0,0 +1,127 @@
+package pipeline
+
+import (
+	"fmt"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/MakeNowJust/heredoc"
+	"github.com/buildkite/cli/v3/internal/io"
+	"github.com/buildkite/cli/v3/pkg/cmd/factory"
+	"github.com/buildkite/go-buildkite/v3/buildkite"
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/lipgloss"
+	"github.com/spf13/cobra"
+)
+
+func NewCmdPipelineCreate(f *factory.Factory) *cobra.Command {
+
+	cmd := cobra.Command{
+		DisableFlagsInUseLine: true,
+		Use:                   "create",
+		Short:                 "Creates a new pipeline",
+		Args:                  cobra.MaximumNArgs(1),
+		Long: heredoc.Doc(`
+			Creates a new pipeline in the current org and outputs the URL to the pipeline.
+
+			The pipeline can be a {pipeline_slug} or in the format {org_slug}/{pipeline_slug}.
+			If the pipeline argument is omitted, the user will be prompted for a name and description.
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var repoURL string
+
+			qs := []*survey.Question{
+				{
+					Name:     "pipeline",
+					Prompt:   &survey.Input{Message: "Name:"},
+					Validate: survey.Required,
+				},
+				{
+					Name:     "description",
+					Prompt:   &survey.Input{Message: "Description:"},
+					Validate: survey.Required,
+				},
+			}
+			answers := struct{ Pipeline, Description string }{}
+			err := survey.Ask(qs, &answers)
+			if err != nil {
+				return err
+			}
+
+			repoURLS := getRepoURLS(f)
+			if len(repoURLS) > 0 {
+				prompt := &survey.Select{
+					Message: "Choose a repository:",
+					Options: repoURLS,
+				}
+				err := survey.AskOne(prompt, &repoURL, survey.WithValidator(survey.Required))
+				if err != nil {
+					return err
+				}
+			} else {
+				err := survey.AskOne(&survey.Input{Message: "Repository URL:"}, &repoURL, survey.WithValidator(survey.Required))
+				if err != nil {
+					return err
+				}
```

Comment:
Should this be wrapped into line 45 so its all done at once?

Reply:
I could not find an option that would prompt if the options provided is empty. But maybe there is a way with bubbletea.

Reply:
Oh yeah it probably doesn't exist in this package. Something we can move to with bubbletea in the future maybe 👍